src/db_state.rs

```diff
@@ -117,6 +122,26 @@ pub struct CoreDbState {
     pub(crate) last_compacted_wal_sst_id: u64,
 }
 
+impl fmt::Display for CoreDbState {
```

Comment:
PTAL and let me know if there's any additional information that would be helpful when printing out the core state

Reply:
I like the idea of using the debug output. I had done that originally and it was formatted in a way that was unreadable but I think that was a problem on my end. Here's a sample output:

```
Manifest {
    core: CoreDbState {
        l0_last_compacted: Some(
            Ulid(
                2086828830169504107423537551710075386,
            ),
        ),
        l0: [
            SsTableHandle {
                id: Compacted(
                    Ulid(
                        2086833967412949180235756782188299391,
                    ),
                ),
                info: SsTableInfoOwned {
                    data: b"$�����'
",
                },
            },
            SsTableHandle {
                id: Compacted(
                    Ulid(
                        2086833911911964747698437416831568951,
                    ),
                ),
                info: SsTableInfoOwned {
                    data: b"$�����'",
                },
            },
        ],
        compacted: [
            SortedRun {
                id: 0,
                ssts: [
                    SsTableHandle {
                        id: Compacted(
                            Ulid(
                                2086828835321177149158298892092957540,
                            ),
                        ),
                        info: SsTableInfoOwned {
                            data: b"$/_\
                                        ��'A�",
                        },
                    },
                ],
            },
        ],
        next_wal_sst_id: 1,
        last_compacted_wal_sst_id: 0,
    },
    writer_epoch: 14,
    compactor_epoch: 14,
}
```

(for context, here is how it looked unformatted which is why I didn't go this route the first time around):
```
Manifest {\n    core: CoreDbState {\n        l0_last_compacted: Some(\n            Ulid(\n                2086828830169504107423537551710075386,\n            ),\n        ),\n        l0: [\n            SsTableHandle {\n                id: Compacted(\n                    Ulid(\n                        2086833967412949180235756782188299391,\n                    ),\n                ),\n                info: SsTableInfoOwned {\n                    data: b\"\\x18\\0\\0\\0\\0\\0\\0\\0\\0\\0\\x0e\\0(\\0\\x04\\0\\x08\\0\\x10\\0\\x18\\0 \\0\\x0e\\0\\0\\0$\\0\\0\\0\\xc9\\x07\\x02\\x04\\0\\0\\0\\0\\xd0\\x7f\\x03\\0\\0\\0\\0\\0\\xca\\xdf\\x01\\x04\\0\\0\\0\\0\\xff'\\0\\0\\0\\0\\0\\0\\x04\\0\\0\\0\\0\\0\\0\\n\",\n                },\n            },\n            SsTableHandle {\n                id: Compacted(\n                    Ulid(\n                        2086833911911964747698437416831568951,\n                    ),\n                ),\n                info: SsTableInfoOwned {\n                    data: b\"\\x18\\0\\0\\0\\0\\0\\0\\0\\0\\0\\x0e\\0(\\0\\x04\\0\\x08\\0\\x10\\0\\x18\\0 \\0\\x0e\\0\\0\\0$\\0\\0\\0\\xc9\\x07\\x02\\x04\\0\\0\\0\\0\\xd0\\x7f\\x03\\0\\0\\0\\0\\0\\xca\\xdf\\x01\\x04\\0\\0\\0\\0\\xff'\\0\\0\\0\\0\\0\\0\\x04\\0\\0\\0\\0\\0\\0\\x04\",\n                },\n            },\n        ],\n        compacted: [\n            SortedRun {\n                id: 0,\n                ssts: [\n                    SsTableHandle {\n                        id: Compacted(\n                            Ulid(\n                                2086828835321177149158298892092957540,\n                            ),\n                        ),\n                        info: SsTableInfoOwned {\n                            data: b\"\\x18\\0\\0\\0\\0\\0\\0\\0\\0\\0\\x0e\\0(\\0\\x04\\0\\x08\\0\\x10\\0\\x18\\0 \\0\\x0e\\0\\0\\0$\\0\\0\\0/_(\\x0e\\0\\0\\0\\0\\x08\\\\\\x0c\\0\\0\\0\\0\\0\\xee\\xd1'\\x0e\\0\\0\\0\\0A\\x8d\\0\\0\\0\\0\\0\\0\\x04\\0\\0\\0\\0\\0\\0\\x05\",\n                        },\n                    },\n                ],\n            },\n        ],\n        next_wal_sst_id: 1,\n        last_compacted_wal_sst_id: 0,\n    },\n    writer_epoch: 14,\n    compactor_epoch: 14,\n}
```

I'll take a look at JSON and see if it's straightforard to implement. Would be great to have it in an easily parseable format.

Reply:
I'm super happy with the JSON serde module. Nice and easy to implement too. Will go with that:

```
{
  "core": {
    "l0_last_compacted": "01J7M9NRAGXPV967CK920R7EFT",
    "l0": [
      {
        "id": {
          "Compacted": "01J7MDQE553EWN2MV7JM2JJ13Z"
        },
        "info": {
          "data_length": 72
        }
      },
      {
        "id": {
          "Compacted": "01J7MDP1AFRMF7KGK9H3D64K1Q"
        },
        "info": {
          "data_length": 72
        }
      }
    ],
    "compacted": [
      {
        "id": 0,
        "ssts": [
          {
            "id": {
              "Compacted": "01J7M9NWFP9BMW733J4BH5VRV4"
            },
            "info": {
              "data_length": 72
            }
          }
        ]
      }
    ],
    "next_wal_sst_id": 1,
    "last_compacted_wal_sst_id": 0
  },
  "writer_epoch": 14,
  "compactor_epoch": 14
}
```